dask_deltatable/write.py

```diff
@@ -0,0 +1,262 @@
+            configuration,
+            storage_options,
+            partition_filters,
+        )
+    }
+    graph = HighLevelGraph.from_collections(final_name, dsk, dependencies=(written,))
+    return Scalar(graph, final_name, "")
+
+
+def _commit(
+    table,
+    schemas_add_actions_nested,
+    table_uri,
+    schema,
+    mode,
+    partition_by,
+    name,
+    description,
+    configuration,
+    storage_options,
+    partition_filters,
+):
+    schemas = list(flatten(pluck(0, schemas_add_actions_nested)))
+    add_actions = list(flatten(pluck(1, schemas_add_actions_nested)))
+    # TODO: What should the behavior be if the schema is provided? Cast the
+    # data?
+    if schema:
+        schemas.append(schema)
+
+    # TODO: This is applying a potentially stricted schema control than what
+    # Delta requires but if this passes, it should be good to go
+    schema = validate_compatible(schemas)
+    assert schema
+    if table is None:
+        _write_new_deltalake(
+            table_uri,
+            schema,
+            add_actions,
+            mode,
+            partition_by or [],
+            name,
+            description,
+            configuration,
+            storage_options,
+        )
+    else:
+        table._table.create_write_transaction(
+            add_actions,
+            mode,
+            partition_by or [],
+            schema,
+            partition_filters,
+        )
+        table.update_incremental()
+
+
+def _write_partition(
+    df,
+    *,
+    schema,
+    partitioning,
+    current_version,
+    file_options,
+    max_open_files,
+    max_rows_per_file,
+    min_rows_per_group,
+    max_rows_per_group,
+    filesystem,
+    max_partitions,
+) -> tuple[pa.Schema, list[AddAction]]:
+    # TODO: what to do with the schema, if provided
+    data = pa.Table.from_pandas(df)
+    schema = schema or data.schema
+
+    add_actions: list[AddAction] = []
+
+    def visitor(written_file: Any) -> None:
+        path, partition_values = get_partitions_from_path(written_file.path)
+        stats = get_file_stats_from_metadata(written_file.metadata)
+
+        # PyArrow added support for written_file.size in 9.0.0
+        if PYARROW_MAJOR_VERSION >= 9:
+            size = written_file.size
+        else:
+            size = filesystem.get_file_info([path])[0].size
+
+        add_actions.append(
+            AddAction(
+                path,
+                size,
+                partition_values,
+                int(datetime.now().timestamp() * 1000),
+                True,
+                json.dumps(stats, cls=DeltaJSONEncoder),
+            )
+        )
+
+    ds.write_dataset(
+        data,
+        base_dir="/",
+        basename_template=f"{current_version + 1}-{uuid.uuid4()}-{{i}}.parquet",
+        format="parquet",
+        partitioning=partitioning,
+        # It will not accept a schema if using a RBR
+        schema=schema if not isinstance(data, RecordBatchReader) else None,
```

Comment:
Why would `data` be a `RecordBatchReader` if you just created it as `pa.Table.from_pandas` on line 218?